lib/assets/javascripts/cartodb/common/dialogs/delete_column/delete_column_view.js

```diff
@@ -24,9 +24,8 @@ module.exports = BaseDialog.extend({
   },
 
   _initViews: function() {
-
     this.table = this.options.table;
-    this.column = this.options.column;
+    this.column = this.options.table.get('column_aliases')[this.options.column] || this.options.column;
```

Comment:
I have updated the code to better handle a null of undefined `this.options.table.get('column_aliases')` values.  I had not implemented this in the first change because the server always returns an empty object for the column_aliases but you are correct that having an extra check does not hurt and can only help prevent errors.